pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -361,9 +349,14 @@ func (m *migration) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
 		if _, ok := amConfigPerOrg[rule.OrgID]; !ok {
 			m.mg.Logger.Info("no configuration found", "org", rule.OrgID)
 		} else {
-			if err := m.updateReceiverAndRoute(allChannelsPerOrg, defaultChannelsPerOrg, da, rule, amConfigPerOrg[rule.OrgID]); err != nil {
+			route, err := m.createRouteForAlert(rule.UID, da, receiversPerOrg[rule.OrgID], defaultReceiversPerOrg[rule.OrgID])
+			if err != nil {
 				return err
 			}
+
+			if route != nil {
+				amConfigPerOrg[rule.OrgID].AlertmanagerConfig.Route.Routes = append(amConfigPerOrg[rule.OrgID].AlertmanagerConfig.Route.Routes, route)
+			}
```

Comment:
population of AM config is scattered across two files, which, I think, does not help readablility. I wonder if it is possible to put this in one place. I think you can provide the entire slice of dashAlerts to the method `setupAlertmanagerConfigs` that will just return the fully populated map of AM configs that is ready to be saved.